lib/ddtrace/transport/http/adapters/ethon.rb

```diff
@@ -0,0 +1,98 @@
+require 'ddtrace/transport/response'
+require 'ethon'
+
+module Datadog
+  module Transport
+    module HTTP
+      module Adapters
+        # Adapter for Ethon >= 0.8.0.
+        # More memory efficient than Net::HTTP.
+        class Ethon
+          attr_reader \
+            :hostname,
+            :port,
+            :timeout
+
+          DEFAULT_TIMEOUT = 10
```

Comment:
I'm curious why this is different from the current default we have for the `net_http` adapter. Have you considered possibly extracting the default timeout so that it is used by all?

(Also, I would suggest indicating the unit as well -- `DEFAULT_TIMEOUT_SECONDS`; since we're not using a type that encodes the unit, having it in the name avoids "oops I passed timeout to something that said timeout but actually one of them was seconds and the other was milliseconds" ;_; )